scripts/ci/run_ci.sh

```diff
@@ -269,6 +269,10 @@ if [ -n "$main_ci" ]; then
 	cat test_file_3.txt test_file_2_in.txt test_file_1_in.txt > test_file.txt
 
 	# Run a subset of tests based on matrix size
+    if [[ $SC == *"sanitycheck"* ]]; then
```

Comment:
I am not following what you are doing here to be honest and why are you changing the what_changed.py file.Your commit message is lacking any information, saying you are changing specific file is not helpful, we can see what files you have changed.

Reply:
Added the information in commit message now. I was trying to see via this draft PR if this change passes shippable check. String "sanitycheck" will be printed in what_changed.py and collected in SC variable in run-ci.sh when there are changes to sanitycheck script or sanitylib.py. 
This logic is based on SC string.

Reply:
the sanitycheck tests are cheap to run, we should just run them eveytime and not based on changes. What happens when you change the test data?

Reply:
These changes are in addition to running the zephyr tests. But when string "sanitycheck" is present, then sanitycheck testsuite will also run. As of now pytest runs these tests only on any changes to scripts/sanitycheck or scripts/sanitylib.py. This is because right now test data keeps on changing due to active sanitycheck testsuite development. Do we want to run these tests on any changes to test data or do we want to do that later?

Reply:
> Do we want to run these tests on any changes to test data or do we want to do that later?

Yes, the point of the test is to catch issues also that are not related to sanitycheck changes, for example updated python or some other dependencies of sanitycheck. So please remove this logic and just run it.

Reply:
Removed that logic. Testsuite will now run whatsoever.